org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/OurSyntaxDocument.java

```diff
@@ -312,16 +370,93 @@ private final void do_update(int line) throws BadLocationException {
 	 * Re-color the given line assuming it starts with a given comment mode,
 	 * then return the comment mode for start of next line.
 	 */
+
 	private final int do_reapply(int comment, final String txt, final int line) {
 		while (line >= comments.size())
 			comments.add(-1); // enlarge array if needed
 		comments.set(line, comment); // record the fact that this line starts
 										// with the given comment mode
-		for (int n = txt.length(), i = do_getLineStartOffset(line); i < n;) {
+		int startOfLine = do_getLineStartOffset(line);
+		int endOfLine = txt.indexOf('\n', startOfLine);
+
+		if (comment == 0 && line == 0 && match(txt, 0, "---\n")) {
+			setCharacterAttributes(startOfLine, 4, yamlHeaderBars, false);
+			return 3;
```

Comment:
It would be good to extract this magic "comment mode" numbers into constants

Reply:
I do not like to use constants when they are only used in one place. Just creates a useless indirection and is more error prone. When they are used in multiple places I agree.